server/src/core/odoo.rs

```diff
@@ -926,146 +920,68 @@ impl Odoo {
-                        }
-                    },
-                    "fileCache" => {
-                        if let Some(file_cache) = value.as_bool() {
-                            _file_cache = file_cache
-                        } else {
-                            session.log_message(MessageType::ERROR, String::from("Unable to parse file_cache"));
-                        }
-                    },
-                    "diagMissingImportLevel" => {
-                        if let Some(diag_import_level) = value.as_str() {
-                            _diag_missing_imports = match DiagMissingImportsMode::from_str(diag_import_level) {
-                                Ok(mode) => mode,
-                                Err(_) => {
-                                    session.log_message(MessageType::ERROR, String::from("Unable to parse diag_import_level. Setting it to all"));
-                                    DiagMissingImportsMode::All
-                                }
-                            };
-                        }
-                    },
-                    "configurations" => {
-                        if let Some(values)= value.as_object() {
-                            configurations = values.clone();
-                        }
-                    },
-                    "selectedConfiguration" => {
-                        if let Some(value_str) = value.as_str() {
-                            selected_configuration = value_str.to_string();
-                        }
-                    },
-                    "serverLogLevel" | "disablePythonLanguageServerPopup" => {
-                        // Too late, set it with command line
-                        // disablePythonLanguageServerPopup, does not affect us, pass
-                    },
-                    _ => {
-                        session.log_message(MessageType::ERROR, format!("Unknown config key: {}", key));
-                    },
-                }
+            return Err(S!("No config found for Odoo"));
+        }
+        let value = config
+            .and_then(|c| c.as_object())
+            .and_then(|c| c.get("selectedConfiguration"))
+            .and_then(|v| v.as_str())
+            .map(|v| v.to_string());
+        return Ok(value);
+    }
+
+    pub fn send_all_configurations(session: &mut SessionInfo) {
+        if let Some(ref config_file) = session.sync_odoo.config_file {
+            let mut configs_map = serde_json::Map::new();
+            for entry in &config_file.config {
+                let html = crate::core::config::ConfigFile { config: vec![entry.clone()] }.to_html_string();
+                configs_map.insert(entry.name.clone(), serde_json::Value::String(html));
             }
+
+            configs_map.insert(
+                "__all__".to_string(),
+                serde_json::Value::String(config_file.to_html_string())
+            );
+            session.send_notification(
+                "$Odoo/setConfiguration",
```

Comment:
New custom messages should be documented here: https://github.com/odoo/odoo-ls/wiki/Implement-a-custom-tool-with-the-server